tests/core/policies/test_rule_policy.py

```diff
@@ -93,8 +93,37 @@ def _form_activation_rule(
     )
 
 
+def test_restrict_multiple_user_inputs_in_rules():
+    domain = Domain.from_yaml(
+        f"""
+intents:
+- {GREET_INTENT_NAME}
+actions:
+- {UTTER_GREET_ACTION}
+    """
+    )
+    policy = RulePolicy(restrict_rules=True)
+    greet_events = [
+        UserUttered(intent={"name": GREET_INTENT_NAME}),
+        ActionExecuted(UTTER_GREET_ACTION),
+        ActionExecuted(ACTION_LISTEN_NAME),
+    ]
+
+    forbidden_rule = DialogueStateTracker.from_events(
+        "bla",
+        evts=[
+            ActionExecuted(RULE_SNIPPET_ACTION_NAME),
+            ActionExecuted(ACTION_LISTEN_NAME),
+        ]
+        + greet_events * (policy.ALLOWED_NUMBER_OF_USER_INPUTS + 1),
+    )
+    forbidden_rule.is_rule_tracker = True
+    with pytest.raises(InvalidRule):
+        policy.train([forbidden_rule], domain, RegexInterpreter())
+
+
 def test_rule_policy_has_max_history_none():
-    policy = RulePolicy()
+    policy = RulePolicy(restrict_rules=False)
```

Comment:
why is this explicitly set to `False`?

Reply:
I just set it to `False` in all our tests, so that if we change restriction, it'll not break other tests that don't test for it

Reply:
I would not do that. If we change restrictions and tests break then this might provide valuable insight. Also I think it's weird to test a policy mainly in some sort of "experimental mode" .

Reply:
ok, will update tests, and put restriction false in only one of them with comment